### PR TITLE
CLI Errata refactoring

### PR DESCRIPTION
**Pull request by ldjebran**
```bash
============================================== 204 passed in 2.30 seconds ==============================================
"!!! Congratulations your changes are good to fly, make a great PR! dlezz++ !!!"
(2.7) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/cli/test_errata.py -v -k "ErrataTestCase"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.12, pytest-3.0.3, py-1.4.31, pluggy-0.4.0 -- /home/dlezz/bin/redhat/python/2.7/bin/python
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile: 
plugins: cov-2.4.0, xdist-1.15.0
collected 34 items 

tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_affected_chosts <- ../../bin/redhat/python/2.7/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_affected_chosts_by_erratum_restrict_flag <- ../../bin/redhat/python/2.7/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_cve <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_id PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_id_and_sort_by_issued_date PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_id_and_sort_by_updated_date PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_label PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_label_and_sort_by_issued_date PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_label_and_sort_by_updated_date PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_name PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_name_and_sort_by_issued_date PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_name_and_sort_by_updated_date PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_id PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_id_and_org_id PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_id_and_org_label PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_id_and_org_name PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_name <- ../../bin/redhat/python/2.7/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_name_and_org_id PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_name_and_org_label PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_name_and_org_name PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_sort_by_issued_date PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_user_permission PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_view_available_count_in_affected_chosts <- ../../bin/redhat/python/2.7/lib/python2.7/site-packages/unittest2/case.py SKIPPED

================================================= 11 tests deselected ==================================================
================================ 19 passed, 4 skipped, 11 deselected in 972.58 seconds =================================
```
**_The logic has not changed, changes affected deprecated asserts, variables names  and constants definition_**

### CODE REVIEW

File: tests/foreman/cli/test_errata.py [hunk 1]
```diff
@@ -282,60 +286,69 @@ class ErrataTestCase(CLITestCase):
...
-           1- Create an org with one custom product & repository
-
-           2- Create an org2 with an other custom product & repository
-
-           3- Create an org3 with two other products, each containing one
-              custom repository
+           1. Create an org with one custom product & repository
+           2. Create an other org with an other custom product & repository
+           3. Create a multi products org with two other products, each
+              containing one custom repository
```
Comment (sghai) [inline tests/foreman/cli/test_errata.py]: 👍

File: tests/foreman/cli/test_errata.py [hunk 2]
```diff
@@ -282,60 +286,69 @@ class ErrataTestCase(CLITestCase):
...
-        Repository.synchronize({'id': cls.org_custom_repo['id']})
-        cls.org2 = make_org()
-        cls.org2_product = make_product(
-            options={'organization-id': cls.org2['id']})
-        cls.org2_custom_repo = make_repository({
+        Repository.synchronize({'id': repo['id']})
+        cls.org_product_erratum_count = FAKE_6_YUM_ERRATUM_COUNT
+        cls.org_product_errata_id = FAKE_2_ERRATA_ID
+        # create an other org
+        cls.org_other = make_org()
```
Comment (sghai) [inline tests/foreman/cli/test_errata.py]: tbh `cls.org_other` doesn't describe its purpose. How this org is different from `cls.org` ? is it just by a product/repo/errata ? can we name it like `errata_org`

File: tests/foreman/cli/test_errata.py [hunk 3]
```diff
@@ -282,60 +286,69 @@ class ErrataTestCase(CLITestCase):
...
+        cls.org_other_product = make_product(
```
Comment (sghai) [inline tests/foreman/cli/test_errata.py]: `cls.errata_org_product`

File: tests/foreman/cli/test_errata.py (hunk 2 above)
Comment (ldjebran) [inline tests/foreman/cli/test_errata.py]: so naming errata_org could give a false information to the user reading the code, as org and org_multi also have erratum

Comment (ldjebran) [inline tests/foreman/cli/test_errata.py]: perhaps it was more simple with org, org2 and org3

Comment (ldjebran) [inline tests/foreman/cli/test_errata.py]: @oshtaier any suggestion

Comment (oshtaier) [inline tests/foreman/cli/test_errata.py]: yeah, I said same words to you yesterday as @sghai told.
Let's me think about it

Comment (ldjebran) [inline tests/foreman/cli/test_errata.py]: @sghai @oshtaier org_concurrent what do you think

**Comment by ldjebran** [comment]
```bash
(2.7) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/cli/test_errata.py -v -k "ErrataTestCase"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.12, pytest-3.0.3, py-1.4.31, pluggy-0.4.0 -- /home/dlezz/bin/redhat/python/2.7/bin/python
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile: 
plugins: cov-2.4.0, xdist-1.15.0
collected 34 items 

tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_affected_chosts <- ../../bin/redhat/python/2.7/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_affected_chosts_by_erratum_restrict_flag <- ../../bin/redhat/python/2.7/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_cve <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_id PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_id_and_sort_by_issued_date PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_id_and_sort_by_updated_date PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_label PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_label_and_sort_by_issued_date PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_label_and_sort_by_updated_date PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_name PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_name_and_sort_by_issued_date PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_org_name_and_sort_by_updated_date PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_id PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_id_and_org_id PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_id_and_org_label PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_id_and_org_name PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_name <- ../../bin/redhat/python/2.7/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_name_and_org_id PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_name_and_org_label PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_filter_by_product_name_and_org_name PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_list_sort_by_issued_date PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_user_permission PASSED
tests/foreman/cli/test_errata.py::ErrataTestCase::test_positive_view_available_count_in_affected_chosts <- ../../bin/redhat/python/2.7/lib/python2.7/site-packages/unittest2/case.py SKIPPED

================================================= 11 tests deselected ==================================================
================================ 19 passed, 4 skipped, 11 deselected in 1154.19 seconds ================================
(2.7) dlezz@elysion:~/projects/robottelo-fork$ 
```

**Comment by ldjebran** [comment]
@abalakh please review if addressed all your comments with this refactoring

**Comment by ldjebran** [comment]
@sghai we have not found better than your naming :)

**Comment by rochacbruno** [comment]
@abalakh do you want to take another look before merging?

**Pull request merged**